test_pull_requests

```diff
@@ -935,6 +969,91 @@ popd
       end
     end
 
+    # has_valid_flake_comment? determines if a valid flake explanation comment exists,
+    # if there exists no valid comment, a human-readable reason why to help users
+    # understand the expected syntax
+    #
+    # A flake comment is valid iff:
+    #  - the comment contains the failed build URL
+    #  - all given issue links in the comment:
+    #    - reside in the correct repository
+    #    - point to issues with the correct label
+    def has_valid_flake_comment?(comments, build_url, settings)
+      general_issue_spec = /(https?:\/\/github.com\/.*\/issues\/[0-9]*)/
+      flake_issue_spec = /(https?:\/\/github.com\/#{Regexp.quote(Properties['github_user'])}\/#{Regexp.quote(settings['flake_identification']['repo'])}\/issues\/([0-9]*))/
+
+      valid_comment_exists = false
+      flake_label = settings['flake_identification']['label']
+      flake_repo = settings['flake_identification']['repo']
+      org_repo = %{#{Properties['github_user']}/#{flake_repo}}
+      flake_label_query = CGI.escape("label:#{flake_label}")
+      issue_link = %{https://github.com/#{org_repo}/issues?q=#{flake_label_query}}
+      reason = %{No comment was found that meets the flake identification requirements. A correctly formatted comment should include a link to the failed Jenkins job and at least one link to a GitHub issue in the [`#{org_repo}` repository](#{issue_link}) with the `#{flake_label}` label. An example of a comment with correct syntax would look like:
+> @#{Properties['bot_github_user']}, the [last build](#{build_url}) failed due to the following flakes:
```

Comment:
Especially in terms of making the repository friendly for people that don't know all the process, the more "human" these comments look like, the better. I want to set a good example. If people leave thinking they *need* to have a well-written comment, I call that a win. If they instead think around it and find the minimal comment they can make and be accepted, whatever, fine.